autorest/authorization_storage_test.go

```diff
@@ -0,0 +1,122 @@
+package autorest
+
+// Copyright 2017 Microsoft Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestNewSharedKeyAuthorizer(t *testing.T) {
+	auth, err := NewSharedKeyAuthorizer("golangrocksonazure", "YmFy", SharedKey)
+	if err != nil {
+		t.Fatalf("create shared key authorizer: %v", err)
+	}
+	req, err := http.NewRequest(http.MethodGet, "https://golangrocksonazure.blob.core.windows.net/some/blob.dat", nil)
+	if err != nil {
+		t.Fatalf("create HTTP request: %v", err)
+	}
+	req.Header.Add("Accept-Charset", "UTF-8")
+	req.Header.Add(headerContentType, "application/json")
+	req.Header.Add(headerXMSDate, "Wed, 23 Sep 2015 16:40:05 GMT")
+	req.Header.Add(headerContentLength, "0")
```

Comment:
potentially pedantry: but I'd suggest this might be better to hard-code these keys to account for potential bugs in the key names?

Reply:
Good point I'll remove the strings, replacing them with constants.